crates/holochain/src/core/workflow/publish_dht_ops_workflow/query.rs

```diff
@@ -0,0 +1,76 @@
+use std::time::SystemTime;
+use std::time::UNIX_EPOCH;
+
+use holo_hash::AgentPubKey;
+use holochain_sqlite::db::ReadManager;
+use holochain_state::query::prelude::*;
+use holochain_types::dht_op::DhtOp;
+use holochain_types::dht_op::DhtOpType;
+use holochain_types::env::EnvRead;
+use holochain_zome_types::Entry;
+use holochain_zome_types::SignedHeader;
+use rusqlite::named_params;
+
+use crate::core::workflow::error::WorkflowResult;
+
+use super::MIN_PUBLISH_INTERVAL;
+
+/// Get all dht ops on an agents chain that need to be published.
+/// - Don't publish private entries.
+/// - Only get ops that haven't been published within the minimum publish interval
+/// - Only get ops that have less then the RECEIPT_BUNDLE_SIZE
+pub fn get_ops_to_publish(
+    agent: &AgentPubKey,
+    env: EnvRead,
+    required_receipt_count: u32,
+) -> WorkflowResult<Vec<DhtOp>> {
+    let earliest_allowed_time = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .ok()
+        .and_then(|epoch| epoch.checked_sub(MIN_PUBLISH_INTERVAL))
+        .map(|t| t.as_secs())
+        .unwrap_or(0);
+    let results = env.conn()?.with_reader(|txn| {
+        let mut stmt = txn.prepare(
+            "
+            SELECT 
+            Header.blob as header_blob,
+            Entry.blob as entry_blob,
+            DhtOp.type as dht_type
+            JOIN
+            DhtOp ON DhtOp.header_hash = Header.hash
+            LEFT JOIN
+            Entry ON (Header.entry_hash IS NULL OR Header.entry_hash = Entry.hash)
```

Comment:
```suggestion
            Entry ON Header.entry_hash = Entry.hash
```

I *think* this would be equivalent since it's a LEFT JOIN and unmatched rows from the left table would be included anyway